test/new-e2e/utils/payloadtesthelper/checkRunAggregator.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package payloadtesthelper
+
+import (
+	"encoding/json"
+)
+
+type checkRun struct {
+	Check     string   `json:"check"`
+	HostName  string   `json:"host_name"`
+	Timestamp int      `json:"timestamp"`
+	Status    int      `json:"status"`
+	Message   string   `json:"message"`
+	Tags      []string `json:"tags"`
+}
+
+func (cr *checkRun) name() string {
+	return cr.Check
+}
+
+func parseCheckRunPayload(data []byte) (checks []*checkRun, err error) {
+	enflated, err := enflate(data)
+	if err != nil {
+		return nil, err
+	}
+	checks = []*checkRun{}
+	err = json.Unmarshal(enflated, &checks)
+	return checks, err
+}
+
+type CheckRunAggregator struct {
```

Comment:
You can further factorised the code by creating a generic `Aggregator[T any]` struct.
you define `newAggregator[T any](parser func(data []byte) (checks []*T, err error))`
`Aggregator` has all the methods from `CheckRunAggregator` but the implementation uses `parser` passed as parameter instead of `parseCheckRunPayload`.

`NewCheckRunAggregator` can be implemented as:
```
func  NewCheckRunAggregator() Aggregator[checkRun] {
   return newAggregator(parseCheckRunPayload)
}
```
Note: `unmarshallPayloads` can be moved to the same as the Aggregator.

Reply:
I'll try having an `Aggregator[P PayloadItem]` 

https://giphy.com/clips/cameronpaulsmith-cat-typing-meme-nuufztgCvyJZIuSkgd